crates/primitives/src/chain/spec.rs

```diff
@@ -278,6 +278,17 @@ impl From<EthersGenesis> for ChainSpec {
             );
         }
 
+        // Time-based hardforks
+        let time_hardfork_opts = vec![
+            (Hardfork::Shanghai, genesis.config.shanghai_time),
+        ];
+        let time_hardforks = time_hardfork_opts
+            .iter()
+            .filter_map(|(hardfork, opt)| opt.map(|time| (*hardfork, ForkCondition::Timestamp(time))))
+            .collect::<BTreeMap<_, _>>();
```

Comment:
since this is only one entry, can we instantiate this via `BTreeMap::from([(_,_)])` instead?

Reply:
because the `shanghai_time` is an option, we want the `BTreeMap` to be empty if the time is `None`, which means we need to do the `filter_map` and map to `ForkCondition::Timestamp`. So I'm not sure how we would use the `BTreeMap::from` in this case?

Reply:
I see,

you can use this, because Option is `IntoIter`

```rust
genesis
.config
.shanghai_time
.map(|time| (Hardfork::Shanghai, ForkCondition::Timestamp(time)))
.into_iter()
.collect::<BTreeMap<_, _>>();
```

Reply:
oh nice! I didn't know `Option` was `IntoIter`

Reply:
til